### ISSUE TITLE
Problem while SetProcessDpiAwarenessContext (starting game impossible)

### ISSUE THREAD

**Issue by divStar** [SteffenL/doa6-resolution-mod#1]
Hey there!

Thank you so much for giving DoA6 a try - and trying to provide a resolution modifier for this one, too! I am sure I'd love it - but unfortunately it doesn't start.

I extracted the DoA6-mod into DoA6-directory in steam and set the ini-file as expected (resolution). However - regardless of whether I set "DpiAware" to 1 or 0, I get the following `mod_resolution.log` output:
```
[debug] Starting up
[info] DOA6 Resolution Mod 0.1.0 by Steffen André langnes
[info] Website: https://www.steffenl.com
[debug] Loading d3d11 from system
[debug] Creating anti-tamper fix
[debug] Setting DPI awareness
[error] Unable to set DPI awareness using SetProcessDpiAwarenessContext
```
I am using a stock DoA6.exe (without any changes).

My `mod_resolution.ini` looks like this (reduced to relevant parts):
```
[Video]
Resolution=5120x2160
FixAspectRatio=1
DpiAware=1

[Proxy]
d3d9=
d3d11=

[Log]
Level=4
File=mod_resolution.log

[Development]
FixAntiDebug=0
SteamAppId=838380
```

Do you have any idea why that might be happening? I'll go and checkout DoA5 mod-resolution 0.5.2, because I haven't yet.

**Comment by divStar** [comment]
Reporting back: DoA5LR mod resolution 0.5.2 works just fine.

```
[Video]
Resolution=5120x2160
FixAspectRatio=1
DpiAware=1
FixViewport=1

[Proxy]
d3d9=

[Log]
Level=3
File=mod_resolution.log
```

mod_resolution.log
```
[debug] Starting up
[info] DOA5LR Resolution Mod 0.5.2 by Steffen André langnes
[info] Website: https://www.steffenl.com
[debug] Setting DPI awareness
[debug] Creating Steam API hook
[debug] Creating aspect ratio patch
[debug] Loading d3d9 from system
[debug] Created game mod object
[debug] Creating viewport patch
[debug] Creating resolution hook
[debug] Shutting down
```

Everything worked as expected. Not sure what's missing in DoA6, but I'd be glad if I didn't have to manually "hack" the resolution into the executable file. So I'll gladly check out and help testing. Sadly my C++ knowledge is nearly 0 (I am fine with many other programming languages, but pointers and direct DLL calls kinda lost me on C++).

**Comment by divStar** [comment]
Okay so .. I am not sure how exactly this setting can prevent the game from starting with your mod, but I figured out the following:

If you have any "HiDPI-scaling override" set to the file, the startup will for some reason fail (this might make a nice addition to the readme).

In particular:
![grafik](https://user-images.githubusercontent.com/138585/87259109-0de74980-c4a9-11ea-83c1-323d3099a19c.png)
In this picture you see "Properties of DOA6.exe" and the child-dialog, resulting from a click of the "Change HiDPI-settings"-button. If the setting within the red rectangle is set, the game will not start up.

As a matter of fact it this setting is unnecessary, so I suppose most people won't have this setting active anyway. But perhaps it'd make a nice "info" in the `mod_resolution_readme.txt` (I suppose this applies to the DOA5LR-mod-resolution as well).

Anyhow: thank you very much!

Btw.: if I mod the `DOA6.exe`-file instead of using your mod, the menu stays in 3840x2160 while in-game I get 5120x2160. However: while in that case I did have some "flickering" artifacts in-game, I don't have those with your approach. Thank you very much!

**Comment by SteffenL** [comment]
Thanks for reporting this! I'll look into the DPI-related error. The fix could be as simple as checking if DPI-awareness is already enabled and/or ignoring the error. The difference from the DOA5LR mod is that the DOA6 mod is using [`SetProcessDpiAwarenessContext`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setprocessdpiawarenesscontext) instead of [`SetThreadDpiAwarenessContext`](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setthreaddpiawarenesscontext) because the game is multithreaded and using the latter function only works for the opening splash screen. Don't know why they would work differently though.

Also please check out the latest release for fixing the stretched menu.

**Comment by divStar** [comment]
Thank you! After removing said check box from the HiDPI properties, DOA6 worked flawlessly!

**Comment by SteffenL** [comment]
Fixed in ab84d6bede1d3fff5fa26634ad6e453242f56957 and released in version 0.1.2. Thank you for the feedback!
